### ISSUE TITLE
Server side filtering

### ISSUE THREAD

**Issue by jfresco** [DemocraciaEnRed/civicstack#19]
- [x] Sort by date, upvotes, and name with basic UI
- [x] Search
- [x] Search widget
- [x] Make filter in server side
- [ ] Design sort buttons on homepage @mjlescano

**Comment by jfresco** [comment]
I implemented MongoDB's text search as [explained here](https://docs.mongodb.com/manual/reference/operator/query/text/#definition), but testing it I've found that it only matches full words. I mean, if I search for `dem` it will not match `democracy`.
I digged in SO and I've found that [the only way to match partial words is by using regex](http://stackoverflow.com/questions/35152310/mongoose-text-search-with-partial-string).
I noticed you're using MongoDB's `$text` in DemocracyOS so I assume you're getting the same behavior.

Is it acceptable or should we go for a regex based solution?
